docs/toolkits/openstack-operations-maintenance.md

```diff
@@ -1,47 +1,181 @@
+
+To add a user, you can run:
+``` bash
+openstack user create --domain <default domain> --password-prompt <user name>
+```
+This will ask you for a password.  You can pass this information to each user to allow them to access the OpenStack cloud.  To delete a user, it's simply:
+``` bash
+openstack user delete <user name>
+```
+
+The newly created user will need a project to operate on as well.  If they don't have a project, you can create one with:
+``` bash
+openstack project create --domain <domain name> --description "<A good description here>" <project name>
+```
+
+In OpenStack, you don't simply add a user to a project.  There needs to be a role associaction created, associating the user and the project.  The role itself can be a priviledged user or standard user.  If you don't have a standard user role created, you can create one with:
+``` bash
+openstack role create <role name>
+```
+Role associations are created with:
+``` bash
+openstack role add --project <project name> --user <user name> <role name>
+```
+
+To remove the role association, run:
+``` bash
+openstack role remove --user <user name> --project <project name> <role name>
+```
 
-Checks for a single node database:
+Oftentimes, different projects or users will have different needs of the OpenStack cloud.  This can lead to the default quotas not allowing them to do something that fits their workflow.  For example, perhaps they need more than the default quota of volumes.  These can be adjusted with the CLI by running:
+``` bash
+openstack quota set --volumes <new volume quota> <project name>
+```
+Note that quotas are set per project and not per user.
+
+
+
+### Maintenance
+
+---
+
+#### NTP
+	- What NTP Errors look like
+		The biggest offender here is going to be any clock drift errors in the logs and odd behavior by computes on some isolated VMs.
+	- How to fix
+		Usually this is either two issues.  First, the time daemon running on your node may simply need a restart.  The second is that it could simply be a misconfiguration on said time daemon.
+
+#### Rabbit MQ
+
+	- Checks:
+		To check the status of your RabbitMQ cluster, run the following on one of the rabbitmq nodes:
+``` bash
+rabbitmqctl cluster_status
+```
+Some things to look for here, are any errors or alarms under the [Alarms] section.
+
+	- What rabbitMQ errors look like
+		rabbitMQ errors can be tough to diagnose.  All the openstack services send messages through rabbitMQ, so there is a broad range of errors that an administrator can come across while trying to determine the cause.
+	- How to fix
+		The simplest fix is usually going to be a dead rabbitMQ service on one of the nodes.  Usually a restart can fix these.
+
+#### Mysql
+
+	- Checks for a single node database:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 Also, you can check the logs to make sure there are no discrepancies.
```

Comment:
```suggestion
#### MySQL

	- Checks for a single node database:
SSH to the MySQL node and log into the database with:
`mysql`
Also, you can check the logs to make sure there are no discrepancies.
```
capitalization

Also, we might want to say something like "MySQL (a.k.a. MariaDB)" to de-confuse new operators who may be using MariaDB instead of MySQL :shrug: